From ec32dc61d2edb0f550099a7be3dac781e6d09096 Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Tue, 27 Jun 2017 09:48:45 +0800
Subject: [PATCH] arm: dts: rk3288-android: enable built-in hdmi and rt5640
 sound

This patch use one cpudai->multicodecs card for built-in hdmi and
external codec which share the same i2s controller.

Change-Id: Ib56d7f10b6739b1c26ec8946470b9f0cd7215979
Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-android.dtsi        | 17 +++++++++++++++++
 arch/arm/boot/dts/rk3288-evb-act8846-edp.dts |  4 ++++
 arch/arm/boot/dts/rk3288-evb-rk808.dts       |  4 ++++
 arch/arm/boot/dts/rk3288-evb-rk818-edp.dts   |  4 ++++
 arch/arm/boot/dts/rk3288-evb-rk818-mipi.dts  |  4 ++++
 arch/arm/boot/dts/rk3288-evb.dtsi            |  3 ++-
 6 files changed, 35 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/rk3288-android.dtsi b/arch/arm/boot/dts/rk3288-android.dtsi
index d3a3ebb467f2..88a915e207a4 100644
--- a/arch/arm/boot/dts/rk3288-android.dtsi
+++ b/arch/arm/boot/dts/rk3288-android.dtsi
@@ -257,6 +257,23 @@
 		clock-names = "clk_nandc", "hclk_nandc";
 		status = "disabled";
 	};
+
+	hdmi_analog_sound: hdmi-analog-sound {
+		status = "disabled";
+		compatible = "rockchip,rk3288-hdmi-analog",
+				"rockchip,rk3368-hdmi-analog";
+		rockchip,cpu = <&i2s>;
+		rockchip,codec = <&rt5640>, <&hdmi>;
+		rockchip,widgets =
+			"Microphone", "Microphone Jack",
+			"Headphone", "Headphone Jack";
+		rockchip,routing =
+			"MIC1", "Microphone Jack",
+			"MIC2", "Microphone Jack",
+			"Microphone Jack", "micbias1",
+			"Headphone Jack", "HPOL",
+			"Headphone Jack", "HPOR";
+	};
 };
 
 &edp_panel {
diff --git a/arch/arm/boot/dts/rk3288-evb-act8846-edp.dts b/arch/arm/boot/dts/rk3288-evb-act8846-edp.dts
index cfa8c382df49..5f5fb3277bb5 100644
--- a/arch/arm/boot/dts/rk3288-evb-act8846-edp.dts
+++ b/arch/arm/boot/dts/rk3288-evb-act8846-edp.dts
@@ -96,6 +96,10 @@
 	mali-supply = <&vdd_gpu>;
 };
 
+&hdmi_analog_sound {
+	status = "okay";
+};
+
 &i2c0 {
 	clock-frequency = <400000>;
 
diff --git a/arch/arm/boot/dts/rk3288-evb-rk808.dts b/arch/arm/boot/dts/rk3288-evb-rk808.dts
index f1347219b906..0d5aec924cc7 100644
--- a/arch/arm/boot/dts/rk3288-evb-rk808.dts
+++ b/arch/arm/boot/dts/rk3288-evb-rk808.dts
@@ -73,6 +73,10 @@
 	cpu0-supply = <&vdd_cpu>;
 };
 
+&hdmi_analog_sound {
+	status = "okay";
+};
+
 &i2c0 {
 	clock-frequency = <400000>;
 
diff --git a/arch/arm/boot/dts/rk3288-evb-rk818-edp.dts b/arch/arm/boot/dts/rk3288-evb-rk818-edp.dts
index 8215545f1683..7f9bcb657fe6 100644
--- a/arch/arm/boot/dts/rk3288-evb-rk818-edp.dts
+++ b/arch/arm/boot/dts/rk3288-evb-rk818-edp.dts
@@ -130,6 +130,10 @@
 	mali-supply = <&vdd_gpu>;
 };
 
+&hdmi_analog_sound {
+	status = "okay";
+};
+
 &i2c0 {
 	clock-frequency = <400000>;
 
diff --git a/arch/arm/boot/dts/rk3288-evb-rk818-mipi.dts b/arch/arm/boot/dts/rk3288-evb-rk818-mipi.dts
index 9e58f7833f49..5b55531ca4aa 100644
--- a/arch/arm/boot/dts/rk3288-evb-rk818-mipi.dts
+++ b/arch/arm/boot/dts/rk3288-evb-rk818-mipi.dts
@@ -92,6 +92,10 @@
 	mali-supply = <&vdd_gpu>;
 };
 
+&hdmi_analog_sound {
+	status = "okay";
+};
+
 &i2c0 {
 	clock-frequency = <400000>;
 
diff --git a/arch/arm/boot/dts/rk3288-evb.dtsi b/arch/arm/boot/dts/rk3288-evb.dtsi
index 3e1f38168292..f8284cc5f78a 100644
--- a/arch/arm/boot/dts/rk3288-evb.dtsi
+++ b/arch/arm/boot/dts/rk3288-evb.dtsi
@@ -59,7 +59,8 @@
 		};
 	};
 
-	sound {
+	sound: sound {
+		status = "disabled";
 		compatible = "simple-audio-card";
 		simple-audio-card,format = "i2s";
 		simple-audio-card,name = "rockchip,rt5640-codec";
-- 
2.35.3

